contracts/src/chainlink/solidity/emergency/StarknetValidator.sol

```
@@ -123,15 +133,15 @@ contract StarknetValidator is TypeAndVersionInterface, AggregatorValidatorInterf
     int256 /* previousAnswer */,
     uint256 /* currentRoundId */,
     int256 currentAnswer
-  ) external override checkAccess returns (bool) {
+  ) external payable override checkAccess returns (bool) {
     return _sendUpdateMessageToL2(currentAnswer);
   }
 
   /**
    * @notice retries to send the latest answer as update message to L2
    * @dev only with access, useful in cases where a previous x-domain message was handeled unsuccessfully.
    */
-  function retry() external checkAccess returns (bool) {
+  function retry() external payable checkAccess returns (bool) {
```

Based on the other L2 validators (and the deposit/withdraw function defined here) I don't think the Aggregator is supposed to fund the fees. The validator is meant to withdraw the fees from it's own balance. So I think to get the tests to pass, just deposit some eth to this contract (it has a payable receive() method)

---

Example on Arbitrum https://github.com/smartcontractkit/chainlink/blob/c8d7632bd817ca993e40862d53be7910ff38d34c/contracts/src/v0.8/dev/ArbitrumValidator.sol

---

Hmm could you take care of this in a follow-up PR? I just want to unblock merging the dependency bumps

---

I can also exclude these changes and let it fail on that case for now